addons/point_of_sale/static/src/js/models.js

```diff
@@ -1698,13 +1702,18 @@ exports.Orderline = Backbone.Model.extend({
         var total_excluded = round_pr(price_unit * quantity, currency_rounding);
         var total_included = total_excluded;
         var base = total_excluded;
-        _(taxes).each(function(tax) {
+        var taxes_mapped = [];
+
             if (!no_map_tax){
-                tax = self._map_tax_fiscal_position(tax);
-            }
-            if (!tax){
-                return;
+            _(taxes).each(function(tax){
+                _(self._map_tax_fiscal_position(tax)).each(function(tax){
+                    taxes_mapped.push(tax);
+                });
+            });
+        } else {
+            taxes_mapped = taxes;
             }
+        _(taxes_mapped).each(function(tax) {
```

Comment:
taxes_mapped has length 0 while there are taxes assigned to the product.

Reply:
@switch87 Do you refer to how this PR solves https://github.com/odoo/odoo/pull/26337 in a better way? Please let me know if I mis-interpreted... Your comment is still vague to me.

Reply:
If I assign a tax percentage to a product in your branch it doesn't get added on the POS. In master it does. Also combining more taxes works on master but not on your branch. And this is what you tried to do, or am I misunderstanding it?
But however, normal tax calculation doesn't seem to work anymore in my tests.

Reply:
Got it. Thanks!
Error in the control flow fixed. Added: https://github.com/odoo/odoo/pull/26415/files#diff-1c14accbe350f7a177b3e81e5846861bR1675